core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -396,17 +396,21 @@ class ByteBufferMessageSet(val buffer: ByteBuffer) extends MessageSet with Loggi
                                                       compactedTopic: Boolean = false,
                                                       messageFormatVersion: Byte = Message.CurrentMagicValue,
                                                       messageTimestampType: TimestampType,
-                                                      messageTimestampDiffMaxMs: Long): (ByteBufferMessageSet, Boolean) = {
+                                                      messageTimestampDiffMaxMs: Long): (ByteBufferMessageSet, Long, Boolean) = {
```

Comment:
A tuple of 3 elements is not great for readability, I think it's time to introduce a case class for the method result.